src/libasr/pass/array_op.cpp

```diff
@@ -353,6 +353,10 @@ class ArrayOpVisitor : public PassUtils::PassVisitor<ArrayOpVisitor>
         tmp_val = const_cast<ASR::expr_t*>(&(x.base));
     }
 
+    void visit_LogicalConstant(const ASR::LogicalConstant_t& x) {
+        tmp_val = const_cast<ASR::expr_t*>(&(x.base));
```

Comment:
Don't cast like this, rather, do:
```c++
tmp_val = ASR::down_cast<ASR::expr_t*>(&x);
```
Would you please mind sending a PR fixing this?

Reply:
Sure, will do!